cmd/agent/app/flags.go

```diff
@@ -24,68 +24,88 @@ import (
 	"flag"
 	"fmt"
 	"strings"
+
+	"github.com/spf13/viper"
+
+	"github.com/uber/jaeger/pkg/metrics"
+)
+
+const (
+	suffixWorkers             = "workers"
+	suffixServerQueueSize     = "server-queue-size"
+	suffixServerMaxPacketSize = "server-max-packet-size"
+	suffixServerHostPort      = "server-host-port"
+	collectorHostPort         = "collector.host-port"
+	httpServerHostPort        = "http-server.host-port"
+	discoveryMinPeers         = "discovery.min-peers"
 )
 
-// Bind binds the agent builder to command line options
-func (b *Builder) Bind(flags *flag.FlagSet) {
-	b.Metrics.Bind(flags)
-	for i := range b.Processors {
-		p := &b.Processors[i]
-		name := "processor." + string(p.Model) + "-" + string(p.Protocol) + "."
-		flags.IntVar(
-			&p.Workers,
-			name+"workers",
-			p.Workers,
-			"how many workers the processor should run")
-		flags.IntVar(
-			&p.Server.QueueSize,
-			name+"server-queue-size",
-			p.Server.QueueSize,
-			"length of the queue for the UDP server")
-		flags.IntVar(
-			&p.Server.MaxPacketSize,
-			name+"server-max-packet-size",
-			p.Server.MaxPacketSize,
-			"max packet size for the UDP server")
-		flags.StringVar(
-			&p.Server.HostPort,
-			name+"server-host-port",
-			p.Server.HostPort,
-			"host:port for the UDP server")
+var processors = []struct {
+	model    model
+	protocol protocol
+	port     string
+}{
+	{model: "zipkin", protocol: "compact", port: ":5775"},
+	{model: "jaeger", protocol: "compact", port: ":6831"},
+	{model: "jaeger", protocol: "binary", port: ":6832"},
+}
+
+// CreateFlags returns a new FlagSet that defines parameters of the Builder.
+func (b *Builder) CreateFlags() *flag.FlagSet {
```

Comment:
I don't know what are the best practices in go, but this does not access builder. Should it be decoupled from the builder? WDYT?